### DISCUSSION TITLE
Solar production while charging from Grid at night

### DISCUSSION THREAD

**Discussion by Gnec220** [wlcrs/huawei_solar#523] [General]
Hello everyone,

I have a huawei setup with battery. When I charge the battery from grid at night. This is considered as solar production. 

I use the energy pv configuration from the wiki section. 
Is someone facing the same problem ? And found a solution?

My configuration looks like below:

`template:
  - sensor:
      - name: energy_pv
        #friendly_name: "PV Energy"
        unique_id: "PV Energy"
        state: "{{ (states('sensor.inverter_total_yield') | float) - (states('sensor.battery_total_discharge') | float) + (states('sensor.battery_total_charge') | float)}}"
        device_class: energy      
        state_class: total
        unit_of_measurement: kWh`

**Comment by thefox69** [comment]
Hi,
just my 2 cents to your question:
_Is someone facing the same problem ?_ Yes, have the same "wrong" figures.
_And found a solution?_ Up to now no solution
But seems logcal to me if we see the  sensor with "+sensor.battery_total_charge'"
Maybe some  experienced guy have a workarround.
I guess with the dynamic prices more an more users would like to charge from the grid

**Comment by Gnec220** [comment]
Hi,

I created a energy sensor from the input power entity. This gives me only
PV produced thus eliminating most problems. That's my workaround.
see attached image.
![image](https://github.com/wlcrs/huawei_solar/assets/97594578/4702e66b-1d98-4d45-a5d9-93bd6e14ae1a)


thefox69 ***@***.***> schrieb am Mi., 8. Nov. 2023, 16:48:

**Comment by rpatricio122** [comment]
I have the same wrong number because I charge at night from the grid, anyone with any suggestion to update the sensor code?
